x-pack/plugins/security_solution/public/resolver/types.ts

```diff
@@ -512,6 +529,16 @@ export interface SideEffectSimulator {
      * Trigger `ResizeObserver` callbacks for `element` and update the mocked value for `getBoundingClientRect`.
      */
     simulateElementResize: (element: Element, contentRect: DOMRect) => void;
+
+    /**
+     * Get the most recently written clipboard text. This is only updated when `confirmTextWrittenToClipboard` is called.
+     */
+    clipboardText: string;
+
+    /**
+     * Call this to resolve the promise returned by `writeText`.
+     */
+    confirmTextWrittenToClipboard: () => void;
```

Comment:
Writing to the clipboard returns a promise. Each time that happens the text is enqueued to be written to the clipboard. Calling this will resolve the next promise in the queue. We should use this to test what happens when the browser hangs when copying to clipboard. This can happen when copying large stuff.